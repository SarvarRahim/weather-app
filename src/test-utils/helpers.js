import { render } from '@testing-library/react';
import React, { useRef } from 'react';
import { BrowserRouter, Router } from 'react-router-dom';
import { Provider as ReduxProvider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { theme } from '@styles/Theme';
import { store } from '@store';
import { Formik } from 'formik';
import { VideoContext } from '@courses/routes/courses-single/course.context';
import { createMemoryHistory } from 'history';

const AllTheProviders = ({ children, formikValues = {} }) => {
  const videoRef = useRef();
  return (
    <BrowserRouter>
      <ReduxProvider store={store}>
        <ThemeProvider theme={theme}>
          <Formik initialValues={formikValues}>
            <VideoContext.Provider value={videoRef}>
              {children}
            </VideoContext.Provider>
          </Formik>
        </ThemeProvider>
      </ReduxProvider>
    </BrowserRouter>
  );
};

export const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  };
};
