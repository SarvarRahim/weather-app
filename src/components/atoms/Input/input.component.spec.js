import { fireEvent, screen } from '@testing-library/react';
import { Input } from './input.component';
import 'jest-styled-components';
import { customRender } from '@test-utils/helpers';

const setup = ({ onChange }) => {
  return customRender(<Input onChange={onChange} name='test-input' value='' />);
};

const handleChange = jest.fn();
const TEST_MESSAGE = 'Test message';

const EVENT = { target: { value: TEST_MESSAGE } };

test('Render Input component', () => {
  const { asFragment } = setup({});
  expect(asFragment()).toMatchSnapshot();
});

test('Should change value', () => {
  setup({ onChange: handleChange });
  const input = screen.getByTestId('input');
  expect(input.value).toBe('');
  fireEvent.change(input, EVENT);
  expect(handleChange).toHaveBeenCalledWith(TEST_MESSAGE);
});
