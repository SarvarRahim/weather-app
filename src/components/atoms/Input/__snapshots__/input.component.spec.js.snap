// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Input component 1`] = `
<DocumentFragment>
  .c2 {
  position: relative;
  z-index: 1;
  width: 100%;
  padding: 11px 16px 11px 16px;
  background-color: #fff;
  border: 2px solid #dadada;
  border-radius: 100px;
  font-family: 'Manrope',sans-serif;
  font-weight: 600;
  font-size: 16px;
  line-height: 22px;
}

.c2::-webkit-input-placeholder {
  font-weight: normal;
  font-size: 16px;
}

.c2::-moz-placeholder {
  font-weight: normal;
  font-size: 16px;
}

.c2:-ms-input-placeholder {
  font-weight: normal;
  font-size: 16px;
}

.c2::placeholder {
  font-weight: normal;
  font-size: 16px;
}

.c0 {
  position: relative;
  z-index: 1;
  width: 100%;
  background-color: #fff;
}

.c1 {
  position: absolute;
  z-index: 99;
  top: 13px;
  left: 16px;
  font-weight: 600;
  font-size: 16px;
  line-height: 22px;
}

.c3 {
  position: absolute;
  z-index: 99;
  top: 13px;
  right: 16px;
  font-weight: 600;
  font-size: 16px;
  line-height: 22px;
}

<div
    class="c0"
  >
    <span
      class="c1"
    />
    <input
      class="c2"
      data-testid="input"
      name="test-input"
      value=""
    />
    <span
      class="c3"
    />
  </div>
</DocumentFragment>
`;
